src/SeqCli/Forwarder/Storage/BufferAppender.cs

```diff
@@ -0,0 +1,164 @@
+// Copyright Datalust Pty Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using SeqCli.Forwarder.Filesystem;
+
+namespace SeqCli.Forwarder.Storage;
+
+/// <summary>
+///     The write-side of a buffer.
+/// </summary>
+public sealed class BufferAppender : IDisposable
+{
+    readonly StoreDirectory _storeDirectory;
+    BufferAppenderChunk? _currentChunk;
+
+    BufferAppender(StoreDirectory storeDirectory)
+    {
+        _storeDirectory = storeDirectory;
+        _currentChunk = null;
+    }
+
+    public void Dispose()
+    {
+        _currentChunk?.Dispose();
+    }
+
+    public static BufferAppender Open(StoreDirectory storeDirectory)
+    {
+        return new BufferAppender(storeDirectory);
+    }
+
+    /// <summary>
+    /// Try write a batch.
+    ///
+    /// This method does not throw.
```

Comment:
Might be good to add more detail around this? Seems like `_currentChunk.Appender.Sync()` would/could throw `IOException`?

Reply:
We catch those